packages/rlc-common/src/buildSamples.ts

```diff
@@ -0,0 +1,835 @@
+    const itemType = schema.typeName ?? schema.type;
+    if (itemType === "string" || (first.typeName ?? first.type) === undefined) {
+      return `"${first}"`;
+    } else if (itemType === "number") {
+      return `${first}`;
+    }
+    return mockParameterTypeValue(
+      first.typeName ?? first.type ?? first,
+      parameterName,
+      schemaMap,
+      path
+    );
+  }
+  const unionType = getUnionType(type);
+  return mockParameterTypeValue(unionType, parameterName, schemaMap, path);
+}
+
+function mockRecordValues(
+  type: string,
+  parameterName: string,
+  schemaMap: Map<string, Schema>,
+  path: Set<string> = new Set()
+) {
+  let recordType;
+  const schema = schemaMap.get(type) as DictionarySchema;
+  if (schema && schema.additionalProperties) {
+    recordType =
+      schema.additionalProperties.typeName ?? schema.additionalProperties.type;
+    if (recordType !== "string" && !schemaMap.has(recordType)) {
+      schemaMap.set(recordType, schema.additionalProperties);
+    }
+  } else {
+    recordType = getRecordType(type);
+  }
+
+  return recordType
+    ? `{"key": ${mockParameterTypeValue(
+        recordType,
+        parameterName,
+        schemaMap,
+        path
+      )}}`
+    : `{}`;
+}
+
+function mockArrayValues(
+  type: string,
+  parameterName: string,
+  schemaMap: Map<string, Schema>,
+  path: Set<string> = new Set()
+) {
+  let arrayType;
+  const schema = schemaMap.get(type) as ArraySchema;
+  if (schema && schema.items) {
+    arrayType = schema.items.typeName ?? schema.items.type;
+    if (arrayType !== "string" && !schemaMap.has(arrayType)) {
+      schemaMap.set(arrayType, schema.items);
+    }
+  } else if (isArrayObject(type)) {
+    arrayType = getArrayObjectType(type);
+  } else if (isNativeArray(type)) {
+    arrayType = getNativeArrayType(type);
+  }
+  const itemValue = arrayType
+    ? mockParameterTypeValue(arrayType, parameterName, schemaMap, path)
+    : undefined;
+  return itemValue ? `[${itemValue}]` : `[]`;
+}
+
+function mockObjectValues(
+  type: string,
+  _parameterName: string,
+  schemaMap: Map<string, Schema>,
+  path: Set<string> = new Set()
+) {
+  if (path.has(type)) {
+    // skip generating if self referenced
+    return `{} as any /**FIXME */`;
+  }
+  path.add(type);
+  // add object's properties
+  const values: string[] = [];
+  const schema = schemaMap.get(type) as ObjectSchema;
+  const allPropNames = new Set<string>();
+  extractObjectProperties(
+    schema?.properties ?? {},
+    schemaMap,
+    allPropNames,
+    values,
+    path
+  );
+
+  // add parents' properties
+  const parents = getAllParents(schema, schemaMap);
+  for (let parent of parents) {
+    extractObjectProperties(
+      (parent as ObjectSchema)?.properties ?? {},
+      schemaMap,
+      allPropNames,
+      values,
+      path
+    );
+  }
+
+  return `{${values.join(", ")}}`;
+}
+
+function getAllParents(
+  schema: ObjectSchema,
+  schemaMap: Map<string, Schema>
```

Comment:
The idea here is 1) trying to get the detailed type from schema and if we can't then 2) trying to get the type from generated type. So we have strong dependency for schemaMap especially for complex type. It would be hard to get the typescript type by `Record<Record<number | string>>` without schema info.

https://github.com/Azure/autorest.typescript/blob/27f6269db6e444b6054ad443860f7c569af4d316/packages/rlc-common/src/buildSamples.ts#L475-L482